### PR TITLE
Added some unit test for some Extension classes.

### PR DESCRIPTION
**Pull request by SamPlusPlus**
I added some unit test for the extension classes. This is hopefully just the start and I can get around to adding more test and upping our code coverage.

The unit test did expose a small bug, and some obvious code duplication which I removed, but I tried not to go to crazy on refactoring just yet, until we have more test.

### CODE REVIEW

File: src/JuliusSweetland.OptiKey.UnitTests/Extensions/PointExtensionsTests.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+using JuliusSweetland.OptiKey.Extensions;
+using NUnit.Framework;
+
+namespace JuliusSweetland.OptiKey.UnitTests.Extensions
+{
+    [TestFixture]
+    public class PointExtensionsTests
+    {
+        [Test]
+        public void TestCalculateCentrePoint()
```
Comment (SamPlusPlus) [inline src/JuliusSweetland.OptiKey.UnitTests/Extensions/PointExtensionsTests.cs]: When unit testing I noticed that PointExtensions.CalculateCentrePoint is using Convert.ToInt32  which rounds to an even number.

 According to https://msdn.microsoft.com/en-us/library/ffdk7eyz.aspx

"If value is halfway between two whole numbers, the even number is returned; that is, 4.5 is converted to 4, and 5.5 is converted to 6."

Not sure if this rounding is enough to throw anything off, but I figured I would mention it.

File: src/JuliusSweetland.OptiKey.UnitTests/Extensions/DoubleExtensionsTests.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+﻿using JuliusSweetland.OptiKey.Extensions;
+using NUnit.Framework;
+
+namespace JuliusSweetland.OptiKey.UnitTests.Extensions
+{
+    [TestFixture]
+    public class DoubleExtensionsTests
+    {
+        [Test]
+        public void TestCoerceToUpperLimit()
```
Comment (SamPlusPlus) [inline src/JuliusSweetland.OptiKey.UnitTests/Extensions/DoubleExtensionsTests.cs]: It is minor but DoubleExtensions.CoerceToUpperLimit and DoubleExtensions.CoerceToLowerLimit could use Math.Min and Math.Max for readability.

File: src/JuliusSweetland.OptiKey/Extensions/CharExtensions.cs [hunk 3]
```diff
@@ -13,10 +13,10 @@ public static char ToUpperAndRemoveDiacritics(this char c)
             var cAsString = c.ToString(CultureInfo.InvariantCulture);
```
Comment (SamPlusPlus) [inline src/JuliusSweetland.OptiKey/Extensions/CharExtensions.cs]: Bug Fix here: cAsString.RemoveDiacritics(); was never actually returning the value. I am not sure this is being using anywhere, but I fixed it anyway.

**Comment by SamPlusPlus** [comment]
Targets #14

File: src/JuliusSweetland.OptiKey.UnitTests/Extensions/PointExtensionsTests.cs (hunk 1 above)
Comment (JuliusSweetland) [inline src/JuliusSweetland.OptiKey.UnitTests/Extensions/PointExtensionsTests.cs]: @SamPlusPlus That's interesting. Rounding to the nearest even number is pretty arbitrary, but I'm not particularly concerned about it as the decision to round in either direction would also be arbitrary. From a practical perspective we're dealing in pixels, so half a pixel is meaningless, and choosing one pixel to the left, right, up, or down shouldn't make much difference. My only concern with explicitly performing a Math.Ceiling would be that the result still needs to be converted to an Int.

If you would like to change this to a Math.Ceiling and then convert to an Int then I'll happily accept the pull request.

File: src/JuliusSweetland.OptiKey.UnitTests/Extensions/DoubleExtensionsTests.cs (hunk 2 above)
Comment (JuliusSweetland) [inline src/JuliusSweetland.OptiKey.UnitTests/Extensions/DoubleExtensionsTests.cs]: True - I'll accept a PR if you'd like to make this change?

File: src/JuliusSweetland.OptiKey/Extensions/CharExtensions.cs (hunk 3 above)
Comment (JuliusSweetland) [inline src/JuliusSweetland.OptiKey/Extensions/CharExtensions.cs]: Thank you - good spot! The whole method is unused as you say so I'll remove it after merging in your changes to clean up the source a tiny bit.

File: src/JuliusSweetland.OptiKey.UnitTests/Extensions/DoubleExtensionsTests.cs (hunk 2 above)
Comment (JuliusSweetland) [inline src/JuliusSweetland.OptiKey.UnitTests/Extensions/DoubleExtensionsTests.cs]: Actually I'll add your suggested change now. Thanks.

File: src/JuliusSweetland.OptiKey.UnitTests/Extensions/PointExtensionsTests.cs (hunk 1 above)
Comment (JuliusSweetland) [inline src/JuliusSweetland.OptiKey.UnitTests/Extensions/PointExtensionsTests.cs]: Actually I'll add your suggested change now. Thanks.

**Pull request merged**